src/rp2_common/hardware_flash/include/hardware/flash.h

```diff
@@ -44,6 +33,23 @@
  * \include flash_program.c
  */
 
+// PICO_CONFIG: PARAM_ASSERTIONS_ENABLED_FLASH, Enable/disable assertions in the flash module, type=bool, default=0, group=hardware_flash
+#ifndef PARAM_ASSERTIONS_ENABLED_FLASH
+#define PARAM_ASSERTIONS_ENABLED_FLASH 0
+#endif
+
+#define FLASH_PAGE_SIZE (1u << 8)
+#define FLASH_SECTOR_SIZE (1u << 12)
+#define FLASH_BLOCK_SIZE (1u << 16)
+
+#define FLASH_UNIQUE_ID_SIZE_BYTES 8
+
+// PICO_CONFIG: PICO_FLASH_SIZE_BYTES, size of primary flash in bytes or -1 for unknown, type=int, default=-1, group=hardware_flash
+#ifndef PICO_FLASH_SIZE_BYTES
+#define PICO_FLASH_SIZE_BYTES -1
```

Comment:
Is -1 the correct default value? E.g. see here in `hardware_flash.c`

```
#ifdef PICO_FLASH_SIZE_BYTES
    hard_assert(flash_offs + count <= PICO_FLASH_SIZE_BYTES);
#endif
```
Seems this would give a mixed sign comparison if the board header doesn't define flash size. Maybe it should be left undefined, or set to 16M?